experimental/dds/tree2/src/test/feature-libraries/sequence-field/sequenceChangeRebaser.spec.ts

```diff
@@ -346,6 +349,161 @@ describe("SequenceField - Rebaser Axioms", () => {
 	});
 });
 
+interface TestState {
+	/**
+	 * The current length of the sequence being edited
+	 */
+	length: number;
+	/**
+	 * The highest index that will be iterated to to generate inserts, deletes, and moves
+	 */
+	maxIndex: number;
+	/**
+	 * An array of node counts to operate on. For instance, passing [1, 3] would generate inserts, moves, and
+	 * deletes that operate on one node at a time and then 3 nodes at a time.
+	 */
+	numNodes: number[];
+}
+
+type SequenceFieldTestState = FieldStateTree<TestState, TestChangeset>;
+
+/**
+ * See {@link ChildStateGenerator}
+ */
+const generateChildStates: ChildStateGenerator<TestState, TestChangeset> = function* (
+	state: SequenceFieldTestState,
+	tagFromIntention: (intention: number) => RevisionTag,
+	mintIntention: () => number,
+): Iterable<SequenceFieldTestState> {
+	const { numNodes, maxIndex } = state.content;
+	const iterationCap = Math.min(maxIndex, state.content.length);
+
+	// Undo the most recent edit
+	if (state.mostRecentEdit !== undefined) {
+		assert(state.parent?.content !== undefined, "Must have parent state to undo");
+		const undoIntention = mintIntention();
+		const invertedEdit = invert(state.mostRecentEdit.changeset);
+		yield {
+			content: state.parent.content,
+			mostRecentEdit: {
+				changeset: tagChange(invertedEdit, tagFromIntention(undoIntention)),
+				intention: undoIntention,
+				description: `Undo:${state.mostRecentEdit.description}`,
+			},
+			parent: state,
+		};
+	}
+
+	for (const nodeCount of numNodes) {
+		for (let i = 0; i <= iterationCap; i++) {
+			// Insert nodeCount nodes
+			const insertIntention = mintIntention();
+			yield {
+				content: {
+					length: state.content.length + nodeCount,
+					maxIndex,
+					numNodes,
+				},
+				mostRecentEdit: {
+					changeset: tagChange(
+						Change.insert(i, nodeCount),
+						tagFromIntention(insertIntention),
+					),
+					intention: insertIntention,
+					description: `Insert${nodeCount}${nodeCount === 1 ? "Node" : "Nodes"}At${i}`,
+				},
+				parent: state,
+			};
+
+			// Delete nodeCount nodes
+			const deleteIntention = mintIntention();
+			yield {
+				content: {
+					length: state.content.length - nodeCount,
+					maxIndex,
+					numNodes,
+				},
+				mostRecentEdit: {
+					changeset: tagChange(
+						Change.delete(i, nodeCount),
```

Comment:
We avoid generating edits that remove nodes beyond the current advertised `length`.
It technically doesn't matter right now: there's nothing invalid about the edit itsef. This means we could totally ignore the length for this PR (which would be like pretending the sequence started with infinitely many nodes), but we'll probably want to start keeping track the sequence contents in the future, so having the length is a good starting point.

Reply:
Same thing for both moves.

Reply:
I've added some checks. Let me know if they're what you had in mind

Reply:
LGTM!